nucypher/policy/conditions/evm.py

```diff
@@ -163,9 +160,10 @@ def _configure_provider(self, providers: Dict[int, BaseProvider]):
         try:
             provider = providers[self.chain]
         except KeyError:
-            # TODO Use a custom exception class, and catch bubble it up to include info about the node
-            # QUESTION Are nodes required to provide connections to all providers?
-            raise Exception(f'This node does not have a connection to chain {self.chain}')
+            raise self.NoConnectionToChain(
```

Comment:
This exception can be relocated to `nucypher.policy.conditions.exceptions`.

Reply:
Good find. It was relocated, but I need to fix the `self.`